### PR TITLE
Add Proposed OpenTelemetry Trace Contract for PromptFlow Project

### PR DESCRIPTION
**Pull request by liucheng-ms**
# Description

This PR introduces two JSON files demonstrating the proposed contract for integrating OpenTelemetry (OTel) tracing into the PromptFlow project. 

Please refer to [this document](https://opentelemetry.io/docs/concepts/signals/traces/) for general knowledge of the span object.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/sample_trace.json [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        "context": {
+            "trace_id": "0xa0f1ba62b17f09c0cc6fb6a176ed6b77",
+            "span_id": "0x05df0e6a82ee2127",
+            "trace_state": "[]"
+        },
+        "kind": "SpanKind.INTERNAL",
+        "parent_id": "0xa9bf586cf9fba31d",
+        "start_time": "2024-01-22T14:26:57.338927Z",
+        "end_time": "2024-01-22T14:26:57.339127Z",
+        "status": {
```
Comment (JYC-99) [inline src/promptflow/sample_trace.json:13]: Can we align with the existing RH status enum contract for the status?

File: src/promptflow/sample_trace.json [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    },
+    {
+        "name": "my_flow",
+        "context": {
+            "trace_id": "0xa0f1ba62b17f09c0cc6fb6a176ed6b77",
+            "span_id": "0x764a9f505d713128",
+            "trace_state": "[]"
+        },
+        "kind": "SpanKind.INTERNAL",
+        "parent_id": null,
```
Comment (JYC-99) [inline src/promptflow/sample_trace.json:267]: will there only 1 trave with parent id "null" in this json?

Comment (JYC-99) [reply src/promptflow/sample_trace.json:267]: UX needs to pick the information to render the top level overview information above the traces list.
image

File: src/promptflow/sample_trace.json [hunk 3]
```diff
@@ -0,0 +1,290 @@
+[
```
Comment (JYC-99) [inline src/promptflow/sample_trace.json:1]: I do not see samples for LLM calls. Can you please add some and UX can start working on the LLM specific visualization
image

File: src/promptflow/sample_trace.json [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        }
+    },
+    {
+        "name": "number",
+        "context": {
+            "trace_id": "0xa0f1ba62b17f09c0cc6fb6a176ed6b77",
+            "span_id": "0x0c7d8d51aeb79f3b",
+            "trace_state": "[]"
+        },
+        "kind": "SpanKind.INTERNAL",
```
Comment (wangchao1230) [inline src/promptflow/sample_trace.json:102]: why do we need to have a prefix SpanKind?

File: src/promptflow/sample_trace.json [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        "parent_id": "0xa9bf586cf9fba31d",
+        "start_time": "2024-01-22T14:26:57.338927Z",
+        "end_time": "2024-01-22T14:26:57.339127Z",
+        "status": {
+            "status_code": "OK"
+        },
+        "attributes": {
+            "framework": "promptflow",
+            "span_type": "promptflow.Function",
```
Comment (wangchao1230) [inline src/promptflow/sample_trace.json:18]: what are the possible values for span_type?

File: src/promptflow/sample_trace.json [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+            "trace_id": "0xa0f1ba62b17f09c0cc6fb6a176ed6b77",
+            "span_id": "0x05df0e6a82ee2127",
+            "trace_state": "[]"
+        },
+        "kind": "SpanKind.INTERNAL",
+        "parent_id": "0xa9bf586cf9fba31d",
+        "start_time": "2024-01-22T14:26:57.338927Z",
+        "end_time": "2024-01-22T14:26:57.339127Z",
+        "status": {
+            "status_code": "OK"
```
Comment (wangchao1230) [inline src/promptflow/sample_trace.json:14]: I think in OpenTelemetry design this status_code field is optional? If not present, it means OK?

**Pull request closed without merging**